core/unit_test/TestViewAPI.hpp

```diff
@@ -869,8 +869,7 @@ struct TestViewMirror {
 
     Kokkos::parallel_for(
         Kokkos::RangePolicy<typename DeviceType::execution_space>(0, int(10)),
-        // decltype required for Intel classics, that doesn't recognize the CTAD
-        CopyUnInit<decltype(a_d)>(a_d));
+        CopyUnInit(a_d));
```

Comment:
It sounds like the issue is the comment...
```
In file included from /Users/runner/work/kokkos/kokkos/core/unit_test/TestViewAPI_b.hpp:17:
/Users/runner/work/kokkos/kokkos/core/unit_test/TestViewAPI.hpp:872:9: error: no viable constructor or deduction guide for deduction of template arguments of 'CopyUnInit'
        CopyUnInit(a_d));
        ^

Reply:
*shrug* emoji ... 

Just put the template parameter in again.